openedx/core/djangoapps/xblock/runtime/learning_core_runtime.py

```diff
@@ -449,9 +450,20 @@ def _lookup_asset_url(self, block: XBlock, asset_path: str) -> str | None:
                 .get(key=f"static/{asset_path}")
             )
         except ObjectDoesNotExist:
-            # This means we see a path that _looks_ like it should be a static
-            # asset for this Component, but that static asset doesn't really
-            # exist.
-            return None
+            try:
+                # Retry with unquoted path. We don't always unquote because it would not
+                # be backwards-compatible, but we need to try both.
+                asset_path = unquote(asset_path)
```

Comment:
This is to fix assets with a space / `%20` in the URL right? What would break if we change to always unquote? I'd prefer to be consistent in how we handle things, and if we can quickly backport this to Sumac, it's not too late for a change like that. Nobody is using it in production yet.

CC @ormsbee

Reply:
I agree that we should prioritize simplicity/consistency over compatibility given that nobody's running this in production. My main question is where this translation happens. I would have expected this unquoting to happen higher up in the Django request processing–though I don't remember the specifics of how this is called, so maybe there's something preventing that? I just want to make sure that we're quoting in one place and unquoting in one place during request processing, and that we're not throwing around a bunch of paths that may or may not be quoted at various layers of the code.

Reply:
I'm testing this now. Turns out it still fails to render the images in the LMS. What do you guys think about splitting the whitespace-fixing code into another PR so I can test that further without affecting the "xblock-" fix?

Reply:
Yes, please open a separate PR for that.